salt/metalk8s/orchestrate/deploy_etcd_on_new_node.sls

```diff
@@ -0,0 +1,66 @@
+{%- set control_plane_ips = salt.saltutil.runner('mine.get', tgt='*', fun='control_plane_ips') %}
+
+{%- if pillar['bootstrap_id'] in control_plane_ips.keys() and control_plane_ips[pillar['bootstrap_id']] %}
+{%- set bootstrap_ip = control_plane_ips[pillar['bootstrap_id']][0] %}
+{%- else %}
+{%- set bootstrap_ip = 'localhost' %}
+{%- endif %}
+
+{%- if pillar['node_name'] in control_plane_ips.keys() and control_plane_ips[pillar['node_name']] %}
+{%- set node_ip = control_plane_ips[pillar['node_name']][0] %}
+{%- else %}
+{%- set node_ip = 'localhost' %}
+{%- endif %}
+
+{#- etcd endpoint of the new node. #}
+{%- set endpoint  = 'https://' ~ node_ip ~ ':2380' %}
+
+Generate etcd certificates on a new node:
+  salt.state:
+    - tgt: {{ pillar['node_name'] }}
+    - saltenv: {{ saltenv }}
+    - sls:
+      - metalk8s.kubeadm.init.certs.etcd-server
+      - metalk8s.kubeadm.init.certs.etcd-peer
+      - metalk8s.kubeadm.init.certs.etcd-healthcheck-client
+    - pillar:
+        repo:
+          online_mode: false
+          local_mode: false
+          host: {{ bootstrap_ip }}
+        registry_ip: {{ bootstrap_ip }}
+
+Refresh the list of etcd nodes:
+  module.run:
+    - mine.update:
+      - mine_functions: metalk8s.get_etcd_endpoint
```

Comment:
This will run the `mine.update` on the master which is basically not what we want.

`mine.update` it's a minion that push information to the master (NOT master who get information from all minions) so you should call mine.update on all salt-minions.

Something like that
```
Refresh the list of etcd nodes:
  salt.function:
    - name: mine.update
    - tgt: '*'
```

`mine_functions` is only to update only specific informations maybe we don't care here.
Me personnaly I don't care update all mine or only the one you need as you want :)

Reply:
I think we're trying to fix an issue we shouldn't have in the first place because the mine is not the right place to figure out the layout of the etcd cluster (cfr. #877).

Reply:
Ha that may explain why it wasn't working, thanks for the explanation @TeddyAndrieux.